src/CMakeLists.txt

```diff
@@ -787,13 +787,27 @@ add_definitions(-DFREERTOS)
 add_definitions(-D__STACK_SIZE=1024)
 add_definitions(-D__HEAP_SIZE=4096)
 
-# NOTE : Add the following defines to enable debug mode of the NRF SDK:
-#add_definitions(-DDEBUG)
-#add_definitions(-DDEBUG_NRF_USER)
-
-if (NOT CMAKE_BUILD_TYPE)
-    set(CMAKE_BUILD_TYPE "Release")
-endif ()
+# Debug configuration
+if (${CMAKE_BUILD_TYPE} STREQUAL "Debug")
+  add_definitions(-DDEBUG)
+  add_definitions(-DDEBUG_NRF_USER)
+
+  # NRF SDK Logging
+  add_definitions(-DNRF_LOG_ENABLED=1)
```

Comment:
Is this a good idea to define constants that might also be defined in other configuration files ([sdk_config.h](https://github.com/InfiniTimeOrg/InfiniTime/blob/develop/src/sdk_config.h) and [syscfg.h](https://github.com/InfiniTimeOrg/InfiniTime/blob/develop/src/libs/mynewt-nimble/porting/nimble/include/syscfg/syscfg.h)) ?
In my experience, that can be confusing, especially when those constants are defined twice with different values. What do you think?

Reply:
As far as I understand these files, they only provide default fallback values for variables which are not explicit defined (note the `ifndef` guards).

Reply:
For Nimble at least I definitely think they should be defined in our files instead of "modifying the lib" as I see it. That way it's easier to see the configuration changes and maybe use upstream. We could remove the copy of sdk_config.h if we did the same with it, since from the 12000 lines only a couple dozen at most are changed as far as I can tell, making it largely redundant.

Reply:
Isn't the `sdk_config.h` file required by the NRF SDK? Does it supply a default one?

Reply:
The file's copied from config/nrf52832/config/sdk_config.h in the SDK. I'm assuming it could be used somehow.

Reply:
Since the SDK specifically looks for a file named `sdk_config`, we could add `config/nrf52832/config/` to the include path, or put just `#include "config/nrf52832/config/sdk_config.h"` into our `sdk_config.h`.

Reply:
> As far as I understand these files, they only provide default fallback values for variables which are not explicit defined (note the ifndef guards).

> For Nimble at least I definitely think they should be defined in our files instead of "modifying the lib" as I see it. That way it's easier to see the configuration changes and maybe use upstream. We could remove the copy of sdk_config.h if we did the same with it, since from the 12000 lines only a couple dozen at most are changed as far as I can tell, making it largely redundant.

Those are good points! I guess we need to ensure that default fallback definitions contain valid values, or to properly override them in this file.

Reply:
Sounds good. Let's work on that in another PR, so as not to delay merging of this PR.